components/hashglobe/src/diagnostic.rs

```diff
@@ -54,17 +60,19 @@ impl<K: Hash + Eq, V, S: BuildHasher> DiagnosticHashMap<K, V, S>
 
     fn verify(&self) {
         let mut position = 0;
-        let mut bad_canary: Option<(usize, *const usize)> = None;
+        let mut count = 0;
+        let mut bad_canary = None;
         for (_,v) in self.map.iter() {
             let canary_ref = &v.0;
+            position += 1;
             if *canary_ref == CANARY {
-                position += 1;
                 continue;
             }
-            bad_canary = Some((*canary_ref, canary_ref));
+            count += 1;
+            bad_canary = Some((*canary_ref, canary_ref, position));
```

Comment:
actually can we have both count and position here? This tells us its location in the list of full buckets as well as its location in the full list.

The latter can  be calculated from the pointer stuff but it's annoying to do so because the buffer layout is tricky.

Reply:
That makes no sense, because we only have the last bad canary sent to the report, and the `count` number at that point would correspond to the last bad canary as well.

Reply:
oh right sorry. my bad.